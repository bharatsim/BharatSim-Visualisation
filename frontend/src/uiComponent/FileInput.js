import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

import { Box, FormHelperText } from '@material-ui/core';

const FileInput = forwardRef(({ onChange, error }, ref) => {
  function onFileInputChange(event) {
    const uploadedFile = event.target.files[0];
    onChange(uploadedFile);
  }

  return (
    <>
      <input
        type="file"
        ref={ref}
        data-testid="input-upload-file"
        accept=".csv"
        onChange={onFileInputChange}
      />
      <Box>{!!error && <FormHelperText error>{error}</FormHelperText>}</Box>
    </>
  );
});

FileInput.defaultProps = {
  error: '',
};

FileInput.propTypes = {
  onChange: PropTypes.func.isRequired,
  error: PropTypes.string,
};

export default FileInput;
