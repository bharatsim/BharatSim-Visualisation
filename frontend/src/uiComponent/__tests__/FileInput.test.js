import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import FileInput from '../FileInput';

describe('<FileInput />', () => {
  it('should match snapshot', () => {
    const { container } = render(<FileInput onChange={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });

  it('should display validation error if selected object is not file', () => {
    const { queryByText } = render(
      <FileInput onChange={jest.fn()} error="Please upload valid csv a file" />,
    );

    expect(queryByText('Please upload valid csv a file')).toBeInTheDocument();
  });

  it('should pass selected file to onChange callback function', () => {
    const onChange = jest.fn();
    const { getByTestId } = render(<FileInput onChange={onChange} />);

    const fileInput = getByTestId(/input-upload-file/);

    fireEvent.change(fileInput, {
      target: {
        files: [{ name: 'test.csv', type: 'text/csv', size: 2123 }],
      },
    });

    expect(onChange).toHaveBeenCalledWith({ name: 'test.csv', type: 'text/csv', size: 2123 });
  });
});
