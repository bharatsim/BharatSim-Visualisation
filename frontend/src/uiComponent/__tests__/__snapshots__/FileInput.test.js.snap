// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileInput /> should match snapshot 1`] = `
<div>
  <input
    accept=".csv"
    data-testid="input-upload-file"
    type="file"
  />
  <div
    class="MuiBox-root MuiBox-root-1"
  />
</div>
`;
